3.2/root/usr/share/container-scripts/mongodb/README.md

```diff
@@ -88,3 +91,38 @@ the values stored in the variables and the actual passwords. Whenever a database
 container starts it will reset the passwords to the values stored in the
 environment variables.
 
+
+Extending image
+---------------------------------
+This image can be extended using [source-to-image](https://github.com/openshift/source-to-image).
+
+For example to build customized MongoDB database image `my-mongodb-centos7` with configuration in `~/image-configuration/` run:
+
+```
+$ s2i build ~/image-configuration/ centos/mongodb-32-centos7 my-mongodb-centos7
+```
+
+The directory passed to `s2i build` can contain these directories:
+- `mongodb-cfg/`
+  - when running `run-mongod` or `run-mongod-replication` commands contained `mongod.conf` file is used for `mongod` configuration
+    - `envsubst` command is run on this file to still allow customization of the image using environmental variables
+    - custom configuration file does not affect name of replica set - it has to be set in `MONGODB_REPLICA_NAME` environment variable
+    - it is not possible to configure SSL using custom configuration file
+
+- `mongodb-pre-init/`
+  - contained shell scripts (`*.sh`) are sourced before `mongod` server is started
+
+- `mongodb-init/`
+  - contained shell scripts (`*.sh`) are sourced when `mongod` server is started
+    - in this phase `mongod` server don't have enabled authentication when running `run-mongod` command (for `run-mongod-replication` command configured users are already created and authentication is enabled)
+
```

Comment:
When sourcing the scripts, some variables are defined and sometimes users may need them in their extension scripts. For example `REPLICATION` to make things differently for replication use case and non-replication scenarios. I think we should be explicit about which variables users may use this way and consider all the other variables (even if available and set) to be an implementation detail only, so users don't depend on them. I'd start with most necessary variables only and add more when needed later.